### PR TITLE
[QC-810] Integration with GrpGeomHelper for GRP access in Tasks

### PR DESCRIPTION
**Pull request by knopers8**
@shahor02 is this what you need? I moved the init and update calls to `TaskRunner`, so the users do not have to do it themselves.

### CODE REVIEW

**Review by Barthelemy — APPROVED** [review]

**Comment by knopers8** [comment]
please do not merge before we have some feedback from @shahor02 or @chiarazampolli

**Comment by Barthelemy** [comment]
@shahor02 @chiarazampolli could you review as well ? thank you very much.

File: Framework/include/QualityControl/RecoRequestSpecs.h [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+//#include <DetectorsBase/GRPGeomHelper.h>
+#include <string>
+
+namespace o2::quality_control::core
+{
+
+// this reflects the GRPGeomRequest struct, but allows us to gather the values and fill the input specs at different stages
+struct GRPGeomRequestSpec {
+  //  o2::base::GRPGeomRequest::GeomRequest geomRequest = None;
+  std::string geomRequest = "None";
```
[the remark this answers is not in the document]
Comment (shahor02) [reply Framework/include/QualityControl/RecoRequestSpecs.h:29]: None means that geometry is not requested to be loaded.

**Review by shahor02 — APPROVED** [review]
Looks fine to me

File: Framework/include/QualityControl/RecoRequestSpecs.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (knopers8) [reply Framework/include/QualityControl/RecoRequestSpecs.h:29]: ok, very good, then I will not even use `GRPGeomRequest` when it is specified to "None".

[the remark this answers is not in the document]
Comment (shahor02) [reply Framework/include/QualityControl/RecoRequestSpecs.h:29]: Not really: the `none` refers to the geometry option, but the user may still need GRPs.

[the remark this answers is not in the document]
Comment (knopers8) [reply Framework/include/QualityControl/RecoRequestSpecs.h:29]: ah ok, then I misunderstood. I will fix that

[the remark this answers is not in the document]
Comment (knopers8) [reply Framework/include/QualityControl/RecoRequestSpecs.h:28]: fixed

**Comment by knopers8** [comment]
I saw most of the tests passing, the failing ones are not related, thus merged.

File: Framework/src/TaskRunnerFactory.cxx [hunk 2]
```diff
@@ -72,6 +73,19 @@ TaskRunnerConfig TaskRunnerFactory::extractConfig(const CommonSpec& globalConfig
...
                       Lifetime::Timer);
 
+  static std::unordered_map<std::string, o2::base::GRPGeomRequest::GeomRequest> const geomRequestFromString = {
+    { "None", o2::base::GRPGeomRequest::GeomRequest::None },
+    { "Aligned", o2::base::GRPGeomRequest::GeomRequest::Aligned },
+    { "Ideal", o2::base::GRPGeomRequest::GeomRequest::Ideal },
+    { "Alignments", o2::base::GRPGeomRequest::GeomRequest::Alignments }
+  };
+  const auto& grp = taskSpec.grpGeomRequestSpec;
+  auto grpGeomRequest = grp.anyRequestEnabled()                                                               //
```
[the remark this answers is not in the document]
Comment (knopers8) [reply Framework/src/TaskRunnerFactory.cxx:83]: oops

**Pull request merged**